include/oneapi/dpl/pstl/histogram_impl.h

```diff
@@ -18,7 +18,10 @@
 
 #include "histogram_extension_defs.h"
 #include "histogram_binhash_utils.h"
+#include "execution_impl.h"
 #include "iterator_impl.h"
+#include "algorithm_impl.h"
```

Comment:
Would it be sufficient to have just forward declarations for other patterns and bricks here, i.e. include `algorithm_fwd.h`? And would it be preferable to do it this way - e.g. to avoid potential circular dependencies?

Reply:
Yes, I believe `algorithm_impl.h` will always be included alongside `histogram_impl.h` from all paths, so we can do just include the forward declarations.